janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphDefaultSchemaMaker.java

```diff
@@ -34,6 +40,39 @@ public Cardinality defaultPropertyCardinality(String key) {
         return Cardinality.SINGLE;
     }
 
+    @Override
+    public PropertyKey makePropertyKey(PropertyKeyMaker factory, Object value) {
+        String name = factory.getName();
+        Class klass = null;
+        if (value instanceof String) {
+            klass = String.class;
+        } else if (value instanceof Character) {
+            klass = Character.class;
+        } else if (value instanceof Boolean) {
+            klass = Boolean.class;
+        } else if (value instanceof Byte) {
+            klass = Byte.class;
+        } else if (value instanceof Short) {
+            klass = Short.class;
+        } else if (value instanceof Integer) {
+            klass = Integer.class;
+        } else if (value instanceof Long) {
+            klass = Long.class;
+        } else if (value instanceof Float) {
+            klass = Float.class;
+        } else if (value instanceof Double) {
+            klass = Double.class;
+        } else if (value instanceof Date) {
+            klass = Date.class;
+        } else if (value instanceof Geoshape) {
+            klass = Geoshape.class;
+        } else if (value instanceof UUID) {
+            klass = UUID.class;
+        } else {
+            klass = Object.class;
```

Comment:
Wondering if it makes sense to extract this class if/else block into a separate protected method. It would be a lot of logic to copy/paste if the implementation wants to extend with a new `instanceof` check.